src/fiatconnect/FiatDetailsScreen.tsx

```diff
@@ -140,31 +151,38 @@ const FiatDetailsScreen = ({ route, navigation }: Props) => {
         implicitBody[param.name] = param.value
       })
 
+      const fiatAccountSchema = quote.getFiatAccountSchema()
       const completeBody = {
         ...body,
         ...implicitBody,
       }
 
-      // await addNewFiatAccount(quote.getProviderBaseUrl(), fiatAccountSchema, completeBody)
-      //   .then((data) => {
-      //     // TODO Tracking here
-      //     dispatch(showMessage(t('fiatDetailsScreen.addFiatAccountSuccess')))
-      //   })
-      //   .catch((error) => {
-      //     // TODO Tracking here
-      //     if (error === FiatConnectError.ResourceExists) {
-      //       dispatch(showError(ErrorMessages.ADD_FIAT_ACCOUNT_RESOURCE_EXIST))
-      //     } else {
-      //       dispatch(showError(t('fiatDetailsScreen.addFiatAccountFailed')))
-      //     }
-      //     Logger.error(TAG, `Error adding fiat account: ${error}`)
-      //   })
-
-      navigate(Screens.FiatConnectReview, {
-        flow,
-        normalizedQuote: quote,
-        fiatAccount: completeBody as FiatAccount,
+      const fiatConnectClient = await quote.getFiatConnectClient()
+      const result = await fiatConnectClient.addFiatAccount({
+        fiatAccountSchema: fiatAccountSchema,
+        data: completeBody as FiatAccountSchemas[typeof fiatAccountSchema],
       })
+      if (result.isOk) {
+        // TODO Tracking here
+        dispatch(showMessage(t('fiatDetailsScreen.addFiatAccountSuccess')))
+        navigate(Screens.FiatConnectReview, {
+          flow,
+          normalizedQuote: quote,
+          fiatAccount: completeBody as FiatAccount,
+        })
+        setTimeout(() => setIsSending(false), 500)
```

Comment:
Without a timeout to remove the spinner, there's a flicker before navigating to the review screen.